### PR TITLE
fix(core): Don't update document state before applying anchor record

### PR DESCRIPTION
**Pull request by stbrody**

### CODE REVIEW

File: packages/core/src/document.ts [hunk 1]
```diff
@@ -612,9 +612,6 @@ class Document extends EventEmitter {
           return
         }
         case 'COMPLETED': {
-          const state = this._doctype.state
```
Comment (stbrody) [inline packages/core/src/document.ts:615]: With this as it was, an HTTP client that is polling for updates to a document can see and emit an update with a new state that says it's anchored, but doesn't actually include the anchor record in its log

Comment (stbrody) [reply packages/core/src/document.ts:615]: This isn't needed because `_handleTip` will take care of updating the state appropriately

**Review by oed — APPROVED** [review]
Not sure how you caught this, but seems reasonable.

**Review by simonovic86 — APPROVED** [review]
lgtm

**Pull request merged**